src/nwb/RecordingContainers.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "Channel.hpp"
+#include "Types.hpp"
+#include "nwb/base/TimeSeries.hpp"
+
+namespace AQNWB::NWB
+{
+
+/**
+ * @brief The RecordingContainers class provides an interface for managing
+ * and holding groups of TimeSeries acquired during a recording.
+ */
+
+class RecordingContainers
+{
+public:
+  /**
+   * @brief Constructor for RecordingContainer class.
+   */
+  RecordingContainers();
+
+  /**
+   * @brief Deleted copy constructor to prevent construction-copying.
+   */
+  RecordingContainers(const RecordingContainers&) = delete;
+
+  /**
+   * @brief Deleted copy assignment operator to prevent copying.
+   */
+  RecordingContainers& operator=(const RecordingContainers&) = delete;
+
+  /**
+   * @brief Destructor for RecordingContainer class.
+   */
+  ~RecordingContainers();
+
+  /**
+   * @brief Adds a TimeSeries object to the container.
+   * @param data The TimeSeries object to add.
+   */
+  void addData(std::unique_ptr<TimeSeries> data);
+
+  /**
+   * @brief Gets the TimeSeries object from the recordingContainers
+   * @param timeseriesInd The index of the timeseries dataset within the group.
+   */
+  TimeSeries* getTimeSeries(const SizeType& timeseriesInd);
```

Comment:
```suggestion
  Container* getContainer(const SizeType& containerInd);
```

Reply:
In addition to lookup by index, it would be nice to also have a way to look up object by name. This could be a separate issue. One option would be to: change  ``std::vector<std::unique_ptr<TimeSeries>> containers;`` to a Boost `multi_index_container` to allow for look-up both based on index of insertion and a name, where the name could either be custom , unique label defined by the user or the path of the object.

Reply:
> I think we probably want this to be Container. While we currently only use TimeSeries in recording, for event-based recordings (and I think also ROI's in ophys) this will be DynamicTable type containers so we should prepare for this. To safe-guard against issues, I'd add a check in writeTimeSeries to make sure the Container uses is actually a TimeSeries

I updated the class to apply to `Container` more generally. For checking whether the type is actually a `TimeSeries` or `ElectricalSeries`, I used dynamic casting since that should return a `nullptr` if the cast is invalid. Let me know if you had a different approach in mind here?

> In addition to lookup by index, it would be nice to also have a way to look up object by name.

I think this is related to the open issue we have to add tags to the recordingContainers so that we can look up by name/user-defined labels - https://github.com/NeurodataWithoutBorders/aqnwb/issues/38

Reply:
> I think this is related to the open issue we have to add tags to the recordingContainers so that we can look up by name/user-defined labels - #38

Agreed. I added a reference to #38 to this thread. 



> , I used dynamic casting since that should return a `nullptr` if the cast is invalid. Let me know if you had a different approach in mind here?

Since this involves dynamic lookup of objects, I think runtime checking with `dynamic_cast` is the right approach. For anything that we can already check at compile time, I would recommend using compile time asserts like ``static_assert(std::is_base_of<Base, T>::value, "T must be derived from Base")`` instead, but since the functions where this comes into play involve dynamic look-up of objects, I don't think compile time asserts are feasible (at least not without restricting functionality). Or in short, "looks good" 👍